### PR TITLE
Differentiate exceptions with missing keys

### PR DESCRIPTION
**Pull request by fcsonline**
I keep seeing many exceptions in a row when the locale file was not loaded. This is giving us a false positive for missing keys.

Right now we are throwing the same exception when we don't find an i18n key or when the translations were not loaded properly.

Adding a new one for this case.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
   "license": "MIT",
   "jest": { "preset": "ts-jest" },
   "dependencies": {
-    "lodash.at": "4.6.0",
+    "lodash": "4.17.15",
```
Comment (franciscoj) [inline package.json:12]: Shouldn't dependencies be a bit less strict so that it doesn't force a specific verion on the user of the lib?

Comment (fcsonline) [reply package.json:12]: You are right. Changing it

Comment (masylum) [reply package.json:12]: do not bloat factorial-i18n with underscore just for `isEmpty.

```jsx
const isEmpty = (obj) => Object.keys(obj).length === 0
```

File: src/index.ts [hunk 2]
```diff
@@ -36,6 +37,10 @@ export default class I18n {
    * Retrieves a key from the translations object.
    */
   getKey (path: string): TranslationObject | string {
+    if (isEmpty(this.translations)) {
+      throw new Error('Translations have not been loaded')
+    }
```
Comment (masylum) [inline src/index.ts:42]: should this be triggered lazily on in `setTranslations`?

File: package.json (hunk 1 above)
Comment (fcsonline) [reply package.json:12]: I was not requiring the whole lodash, I'm requiring `at` and `isEmpty` only. It is almost the same as you did.

File: src/index.ts (hunk 2 above)
Comment (fcsonline) [reply src/index.ts:42]: I don't get what you mean. What I want to ensure with this is that `setTranslations`  was not called.

File: package.json (hunk 1 above)
Comment (masylum) [reply package.json:12]: will tree-shacking work? I remember some time ago it was not working. Please check the output code.

Comment (fcsonline) [reply package.json:12]: Sure, I can do this

File: src/index.ts (hunk 2 above)
Comment (masylum) [reply src/index.ts:42]: got it 👌

File: package.json (hunk 1 above)
Comment (fcsonline) [reply package.json:12]: Another reason is because the `lodash.at` and `lodash.isEmpty` are not updated anymore.

Comment (fcsonline) [reply package.json]: Did you use `https://www.npmjs.com/package/webpack-bundle-analyzer` to do it?

File: src/index.ts [hunk 3]
```diff
@@ -1,4 +1,5 @@
-const at = require('lodash.at')
+const at = require('lodash/at')
+const isEmpty = require('lodash/isEmpty')
```
Comment (marctorrelles) [inline src/index.ts:2]: Why like that? Wouldn't be the same as this?

```
import { at, isEmpty } from 'lodash'
```

Comment (fcsonline) [reply src/index.ts:2]: We need to ensure the tree-shacking works for both cases.

**Pull request merged**